digdag-ui/console.jsx

```diff
@@ -191,6 +202,23 @@ class CodeViewer extends React.Component {
   }
 }
 
+class CodeEditor extends CodeViewer {
```

Comment:
Extending React components can lead to unexpected behaviours (when children never call super), i think a better approach for this would be to make a component that passes down an option props to CodeViewer

What do you think of something like this?
```
diff --git a/digdag-ui/console.jsx b/digdag-ui/console.jsx
index d686ab14..b8b87074 100644
--- a/digdag-ui/console.jsx
+++ b/digdag-ui/console.jsx
@@ -106,6 +106,20 @@ class CodeViewer extends React.Component {
     className: ?string;
     language: string;
     value: string;
+    editorOptions: Object;
+  }
+
+  static defaultProps = {
+    editorOptions: {
+      enableLinking: true,
+      highlightActiveLine: false,
+      readOnly: true,
+      showLineNumbers: false,
+      showGutter: false,
+      showPrintMargin: false,
+      tabSize: 2,
+      useSoftTabs: true
+    }
   }
 
   /* eslint-disable */
@@ -139,16 +153,7 @@ class CodeViewer extends React.Component {
     const Ace = require('brace')
     require('./ace-digdag')
     this._editor = Ace.edit(this.editor)
-    this._editor.setOptions({
-      enableLinking: true,
-      highlightActiveLine: false,
-      readOnly: true,
-      showLineNumbers: false,
-      showGutter: false,
-      showPrintMargin: false,
-      tabSize: 2,
-      useSoftTabs: true
-    })
+    this._editor.setOptions(this.props.editorOptions)
     this._editor.on('click', this.editorClick.bind(this))
     this._updateEditor(this.props)
   }
@@ -173,6 +178,9 @@ class CodeViewer extends React.Component {
     if (nextProps.value !== this.props.value) {
       this._updateEditor(nextProps)
     }
+    if (this._editor && !_.isEqual(nextProps.editorOptions, this.props.editorOptions)) {
+      this._editor.setOptions(this.props.editorOptions)
+    }
   }
 
   componentWillUnmount () {
```